File: engineering/codebase/smartqr.backend-services/SmartQr.Domain/Codes/Content/Text/Models/TextContent.cs
namespace SmartQr.Domain.Codes.Content.Text.Models;

/// <summary>Free-form text baked directly into the symbol.</summary>
public sealed record TextContent : CodeContent
{
    /// <summary>The literal text to encode.</summary>
    public required string Text { get; init; }

    /// <inheritdoc />
    public override string Encode() => Text ?? string.Empty;
}

File: engineering/codebase/smartqr.frontend-services/src/domain/codes/content/models/TextContent.ts
import { ContentType } from "../enums/ContentType";

/** Represents the static free-text content encoded directly in the QR. */
export interface TextContent {
  /** The content-type discriminator. */
  type: typeof ContentType.Text;

  /** The free-text payload encoded in the QR. */
  text: string;
}
